corehq/apps/reports/views.py

```diff
@@ -1809,14 +1810,24 @@ def dispatch(self, request, *args, **kwargs):
     def tableau_server_form(self):
         data = self.request.POST if self.request.method == 'POST' else None
         return TableauServerForm(
-            data, domain=self.domain
+            data, domain=self.domain, user_syncing_config=self.get_user_syncing_config()
         )
 
-    def get_form_kwargs(self):
-        kwargs = super().get_form_kwargs()
-        kwargs['domain'] = self.domain
-        kwargs['initial'] = TableauServer.objects.get_or_create(domain=self.domain)
-        return kwargs
+    def get_user_syncing_config(self):
+        user_syncing_config = {}
```

Comment:
What do you think about making this a dataclass?

Reply:
Does that also use a dataclass? https://github.com/dimagi/commcare-hq/blob/e63ab6c11d76a37762ab7c1d0b20f16d136b4af2/corehq/apps/app_manager/suite_xml/features/case_tiles.py#L56

The main thing I was considering was that it would help see all the fields in `user_syncing_config` as opposed to adding it piece by piece within the function. But the places where keys are added here are pretty compact so I'm fine with as is also.

Reply:
I don't think we should define the dataclass within the `get_user_syncing_config` function. I was proposing doing it outside of the `get_user_syncing_config` function.

But I like your idea of adding a docstring, and with that, I'm okay with dropping the idea of the dataclass.

Reply:
Thanks! I think seeing all the fields of `user_syncing_config` help with readability. The variable names are already pretty self explanatory so the description might not be as helpful, but I think it's fine as is